pkg/snap/util/calico.go

```diff
@@ -34,3 +48,25 @@ func MaybePatchCalicoAutoDetectionMethod(ctx context.Context, s snap.Snap, canRe
 	}
 	return nil
 }
+
+// replaceAfter replaces the first occurrence of target after the first occurrence of "after" with replacement.
+func replaceAfter(s, after, target, replacement string) (string, error) {
+	if !strings.Contains(s, after) {
+		return "", fmt.Errorf("after string not found: '%s'", after)
+	}
+
+	afterIndex := strings.Index(s, after)
+	if afterIndex == -1 {
+		return "", fmt.Errorf("string not found: '%s'", after)
+	}
+
+	targetIndex := strings.Index(s[afterIndex:], target)
+	if targetIndex == -1 {
+		return "", fmt.Errorf("string not found after '%s': '%s'", after, target)
+	}
+
+	endIndex := afterIndex + targetIndex
+
+	result := s[:endIndex] + replacement + s[len(target)+endIndex:]
+	return result, nil
+}
```

Comment:
I am still missing unit tests for this function. In particular, what is the expected behaviour for the following "matrix": (string exists or not, target exists or not).

I am also not a huge fan of the multiple string concatanations we are doing here, but it should be OK as this is not a critical code path.

Reply:
This is a private utility, to test it I made it public which shouldn't matter much.